js/views/modals/Settings/Moderation.js

```diff
@@ -49,24 +47,32 @@ export default class extends baseVw {
   }
 
   save() {
-    const formData = this.getFormData();
-    this.profile.set(formData);
+    /* if the user isn't already a moderator, the status is true, and the confirmation checkboxes
+     aren't checked, show an error */
+
+    const confirmChecked = this.$understandRequirements.prop('checked') &&
+      this.$acceptGuidelines.prop('checked');
+
+    if (!this.profile.get('moderator') &&
+    $('input[name=moderator]:checked').val() === 'true' && !confirmChecked) {
+      this.$moderationConfirmError.removeClass('hide');
+      return;
+    }
 
-    const save = this.profile.save();
+    this.$moderationConfirmError.addClass('hide');
```

Comment:
It's not necessary to put the 'hide' class on, since this view is re-rendered at the end of this method and the template will apply it. You'll save a DOM write by eliminating this statement.